src/Compilers/CSharp/Portable/Binder/Binder_Constraints.cs

```diff
@@ -575,6 +575,7 @@ static bool isValidConstraintType(TypeParameterSymbol typeParameter, TypeConstra
                     case TypeKind.Array:
                     case TypeKind.Pointer:
                     case TypeKind.FunctionPointer:
+                    case TypeKind.Extension:
```

Comment:
> case TypeKind.Extension: [](http://example.com/codeflow?start=19&length=25)

Is it fine to end up with `extension` as a constraint due to type substitution? For example, while overriding a method with generic constraints. I do not think this code would prevent that.

Reply:
I think that's fine. There's a similar loophole with array types.
I'll add a test.